src/mds/Locker.cc

```diff
@@ -3774,20 +3774,28 @@ void Locker::issue_client_lease(CDentry *dn, client_t client,
     now += mdcache->client_lease_durations[pool];
     mdcache->touch_client_lease(l, pool, now);
 
-    LeaseStat e;
-    e.mask = 1 | CEPH_LOCK_DN;  // old and new bit values
-    e.seq = ++l->seq;
-    e.duration_ms = (int)(1000 * mdcache->client_lease_durations[pool]);
-    encode(e, bl);
-    dout(20) << "issue_client_lease seq " << e.seq << " dur " << e.duration_ms << "ms "
+    uint16_t mask = 1 | CEPH_LOCK_DN;  // old and new bit values
+    uint32_t seq = ++l->seq;
+    uint32_t duration_ms = (int)(1000 * mdcache->client_lease_durations[pool]);
+
+    if (session->info.has_feature(CEPHFS_FEATURE_REPLY_ENCODING)) {
```

Comment:
Lots of duplicate encoding code. How about adding `Locker::encode_lease` which is called here and by `Locker::encode_infinite_lease` and `Locker::encode_null_lease`.